buildpacktoml_test.go

```diff
@@ -286,32 +286,71 @@ func testBuildpackTOML(t *testing.T, when spec.G, it spec.S) {
-					h.Mkfile(t,
-						`[[processes]]`+"\n"+
-							`type = "some-type"`+"\n"+
-							`command = "some-cmd"`+"\n"+
+				when("processes", func() {
+					it("should include processes and use the default value that is set", func() {
+						h.Mkfile(t,
 							`[[processes]]`+"\n"+
-							`type = "other-type"`+"\n"+
-							`command = "other-cmd"`+"\n",
-						filepath.Join(appDir, "launch-A-v1.toml"),
-					)
-					br, err := bpTOML.Build(lifecycle.BuildpackPlan{}, config)
-					if err != nil {
-						t.Fatalf("Unexpected error:\n%s\n", err)
-					}
-					if s := cmp.Diff(br, lifecycle.BuildResult{
-						BOM:         nil,
-						Labels:      []lifecycle.Label{},
-						MetRequires: nil,
-						Processes: []launch.Process{
-							{Type: "some-type", Command: "some-cmd", BuildpackID: "A"},
-							{Type: "other-type", Command: "other-cmd", BuildpackID: "A"},
-						},
-						Slices: []layers.Slice{},
-					}); s != "" {
-						t.Fatalf("Unexpected metadata:\n%s\n", s)
-					}
+								`type = "some-type"`+"\n"+
+								`command = "some-cmd"`+"\n"+
+								`default = true`+"\n"+
+								`[[processes]]`+"\n"+
+								`type = "web"`+"\n"+
+								`command = "other-cmd"`+"\n",
+							// default is false and therefore doesn't appear
+							filepath.Join(appDir, "launch-A-v1.toml"),
+						)
+						br, err := bpTOML.Build(lifecycle.BuildpackPlan{}, config)
+						if err != nil {
+							t.Fatalf("Unexpected error:\n%s\n", err)
+						}
+						if s := cmp.Diff(br, lifecycle.BuildResult{
+							BOM:         nil,
+							Labels:      []lifecycle.Label{},
+							MetRequires: nil,
+							Processes: []launch.Process{
+								{Type: "some-type", Command: "some-cmd", BuildpackID: "A", Default: true},
+								{Type: "web", Command: "other-cmd", BuildpackID: "A", Default: false},
+							},
+							Slices: []layers.Slice{},
+						}); s != "" {
+							t.Fatalf("Unexpected metadata:\n%s\n", s)
+						}
+					})
+					when("when there are multiple default=true processes but they are all of the same type", func() {
+						it("should succeed", func() {
+							h.Mkfile(t,
+								`[[processes]]`+"\n"+
+									`type = "some-type"`+"\n"+
+									`command = "some-cmd"`+"\n"+
+									`default = true`+"\n"+
+									`[[processes]]`+"\n"+
+									`type = "other-type"`+"\n"+
+									`command = "other-cmd"`+"\n"+
+									`[[processes]]`+"\n"+
+									`type = "some-type"`+"\n"+
+									`command = "some-other-cmd"`+"\n"+
+									`default = true`+"\n",
+								filepath.Join(appDir, "launch-A-v1.toml"),
+							)
+							br, err := bpTOML.Build(lifecycle.BuildpackPlan{}, config)
+							if err != nil {
+								t.Fatalf("Unexpected error:\n%s\n", err)
+							}
+							if s := cmp.Diff(br, lifecycle.BuildResult{
+								BOM:         nil,
+								Labels:      []lifecycle.Label{},
+								MetRequires: nil,
+								Processes: []launch.Process{
+									{Type: "some-type", Command: "some-cmd", BuildpackID: "A", Default: true},
```

Comment:
Is it expected the process list will have two with the same type as `Default: true`? I kind of figured the last-in-wins would mean the resulting process collection would only ever show a single entry with `Default: true`.

Reply:
This is interesting. We apply the "last wins" algorithm only in the `exporter` so the builder doesn't care about it.
Correct me if I'm wrong but you're saying that the `builder` should just write one process per buildpack if there is a default process in it, right?
My biggest concern about it is that I'm not sure if it's the `builder` responsibility to do that but I'm not 100% sure.

Reply:
Hmm. That is interesting. My expectations were just from reading the proposed change and trying to align that with the test output. 

I kind of feel like it is partly the builder's responsibility, but I think the exporter would still need to do similar validation. Kind of like duplicating client/server validation. I wouldn't be against it existing in both, but as long as it exists in the exporter I'm 👍

Reply:
I think if a single buildpack tries to write `default = true` for more than one process it should be a failure even if the processes have the same type. We could deal with the edge case of a single buildpack writing duplicate processes separate from this PR, since it existed before.

Reply:
@natalieparellano, I like your idea that whenever a buildpack is trying to write `default = true` for more than one process, the builder should fail (no matter if the processes have the same or different type).
@jabrown85, are you good with this decision?

Reply:
I'm 👍 on adding additional validation here.

Reply:
Please see https://github.com/buildpacks/lifecycle/pull/520/commits/badd029b1d958fe32cb82734998d22b65e454cb0.